### PR TITLE
man: pam_system_loadkey additions and fixes

### PR DESCRIPTION
**Pull request by jonathan-conder**
I started using `pam_systemd_loadkey` recently on NixOS unstable, but had to do some extra digging on top of reading the docs. Hopefully this will help others in future.

One change is to add `pam_gnome_keyring.so` to the `auth` section, in addition to `session`. Without this, my keyring remained locked.

The reason is that [`pam_sm_open_session`](https://gitlab.gnome.org/GNOME/gnome-keyring/-/blob/46.1/pam/gkr-pam-module.c?ref_type=tags#L904) only looks at `gkr_system_authtok`. This is set by [`pam_sm_authenticate`](https://gitlab.gnome.org/GNOME/gnome-keyring/-/blob/46.1/pam/gkr-pam-module.c?ref_type=tags#L851) using `PAM_AUTHTOK`, which in turn is set by `pam_systemd_loadkey`.

The other change is to add a table with alternative values for `keyname`. I've tested that `luks2-pin` works when combined with `systemd-cryptenroll --tpm2-with-pin=true`, but haven't tested `fido2-pin` or `tpm2-pin`.

This works for me because I have a relatively strong PIN; those with weak PINs may prefer to encrypt their keyring with the LUKS2 volume key. I've added a note about that but haven't tested it myself. Happy to remove it if anyone suspects it might not work.

### CODE REVIEW

**Comment by jonathan-conder** [comment]
For convenience, here's a screenshot from the HTML:
screenshot

File: man/pam_systemd_loadkey.xml [hunk 1]
```diff
@@ -49,9 +49,44 @@
         <term><varname>keyname=</varname></term>
 
         <listitem><para>Takes a string argument which sets the keyname to read.
-        The default is <literal>cryptsetup</literal>, which is used by
-        <citerefentry><refentrytitle>systemd-cryptsetup@.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>
-        to store LUKS passphrase during boot.</para>
+        The default is <literal>cryptsetup</literal>.
+        During boot, <citerefentry><refentrytitle>systemd-cryptsetup@.service</refentrytitle><manvolnum>8</manvolnum></citerefentry> stores a passphrase or PIN in the keyring.
+        The LUKS2 volume key can also be used, via the <option>link-volume-key</option> option in <citerefentry><refentrytitle>crypttab</refentrytitle><manvolnum>5</manvolnum></citerefentry>.</para>
```
Comment (yuwata) [inline man/pam_systemd_loadkey.xml]: Could you break the long lines? We typically break lines with 109ch, though, not strict, and sometimes longer e.g. the line that has e.g. `<citerefentry>` or so.

**Review by yuwata — CHANGES_REQUESTED** [review]

**Comment by poettering** [comment]
looks good, but please break lines as suggested.

File: man/pam_systemd_loadkey.xml (hunk 1 above)
Comment (jonathan-conder) [reply man/pam_systemd_loadkey.xml]: Sure, should be fixed now. Output looks identical to me. 

The `<citerefentry>` line is still too long but I think that's OK, if I understood you correctly?

**Review by bluca — APPROVED** [review]

**Pull request merged**